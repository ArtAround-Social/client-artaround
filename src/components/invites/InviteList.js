import React from 'react';
import PropTypes from 'prop-types';
import InviteItem from './InviteItem';

function InviteList({ invites }) {
  const inviteElement = invites.map(invitation => (
    <li key={invitation.id}>
      <InviteItem invitation={invitation} />
    </li>
  ));

  return(
    <ul>{inviteElement}</ul>
  );
}

InviteList.propTypes = {
  invites: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired
  }))
};

export default InviteList;
