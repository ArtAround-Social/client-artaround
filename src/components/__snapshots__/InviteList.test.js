import React from 'react';
import { shallow } from 'enzyme';
import InviteList from '../invites/InviteList';

describe('InviteList component', () => {
  it('renders invites list', () => {
    const wrapper = shallow(
      <InviteList
        invites = { [
          { id: 1234, name: 'art gallery' },
          { id: 4321, name: 'markovco gallery' }
        ] }
        handleReview = {() => { }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
