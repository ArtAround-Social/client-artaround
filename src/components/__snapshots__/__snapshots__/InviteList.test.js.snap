// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteList component renders invites list 1`] = `
<ul>
  <li
    key="1234"
  >
    <Invite
      invintation={
        Object {
          "id": 1234,
          "name": "art gallery",
        }
      }
    />
  </li>
  <li
    key="4321"
  >
    <Invite
      invintation={
        Object {
          "id": 4321,
          "name": "markovco gallery",
        }
      }
    />
  </li>
</ul>
`;
